planner_server:
  ros__parameters:
    expected_planner_frequency: 20.0
    planner_plugins: ['GridBased']
    GridBased:
      plugin: 'nav2_navfn_planner/NavfnPlanner'

global_costmap:
  global_costmap:
    ros__parameters:
      # Whether to send full costmap every update, rather than updates. (default: false)
      always_send_full_costmap: True

      # Reference frame. (default: map)
      global_frame: map

      # Frequency to publish costmap to topic. (default 1.0)
      publish_frequency: 10.0

      # Height of costmap (m). (default: 5)
      width: 5

      # Width of costmap (m). (default: 5)
      height: 5

      # Resolution of 1 pixel of the costmap, in meters. (default: 0.1)
      resolution: 0.1

      # Robot base frame. (default: base_link)
      robot_base_frame: agent_3

      # Robot radius to use, if footprint coordinates not provided. (default: 0.1)
      robot_radius: 0.2

      # Whether costmap should roll with robot base frame. (default: False)
      rolling_window: True

      # Costmap update frequency. (default 5.0)
      update_frequency: 10.0

      # Whether when combining costmaps to use the maximum cost or override. (default: False)
      use_maximum: False

      # List of mapped plugin names for parameter namespaces and names. Note: Each plugin namespace
      # defined in this list needs to have a plugin parameter defining the type of plugin to be
      # loaded in the namespace.
      plugins: ["static_layer", "inflation_layer"]
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        # https://navigation.ros.org/configuration/packages/costmap-plugins/static.html

        # QoS settings for map topic. (default: True)
        map_subscribe_transient_local: True

      inflation_layer:
        plugin: "nav2_costmap_2d::InflationLayer"
        # https://navigation.ros.org/configuration/packages/costmap-plugins/inflation.html

        # Whether the inflation_layer plugin is enabled. (default: True)
        enabled: True

        # Radius to inflate costmap around lethal obstacles. (default: 0.55)
        inflation_radius: 0.3

        # Exponential decay factor across inflation radius. (default: 10.0)
        cost_scaling_factor: 5.0

        # Whether to inflate unknown cells as if lethal. (default: False)
        inflate_unknown: False

        # Whether to inflate unknown cells. (default: False)
        inflate_around_unknown: False


controller_server:
  ros__parameters:
    #odom_topic: /pedsim_simulator/agent_3_pose
    use_sim_time: True
    controller_frequency: 20.0
    min_x_velocity_threshold: 0.001
    min_y_velocity_threshold: 0.5
    min_theta_velocity_threshold: 0.001
    progress_checker_plugin: "progress_checker"
    goal_checker_plugins: ["goal_checker"]
    controller_plugins: ["FollowPath"]

    # Progress checker parameters
    progress_checker:
      plugin: "nav2_controller::SimpleProgressChecker"
      required_movement_radius: 0.5
      movement_time_allowance: 10.0
    # Goal checker parameters
    goal_checker:
      plugin: "nav2_controller::SimpleGoalChecker"
      xy_goal_tolerance: 0.25
      yaw_goal_tolerance: 0.25
      stateful: True
    # DWB parameters
    FollowPath:
      plugin: "dwb_core::DWBLocalPlanner"
      debug_trajectory_details: True
      min_vel_x: 0.0
      min_vel_y: 0.0
      max_vel_x: 0.26
      max_vel_y: 0.0
      max_vel_theta: 1.0
      min_speed_xy: 0.0
      max_speed_xy: 0.26
      min_speed_theta: 0.0
      # Add high threshold velocity for turtlebot 3 issue.
      # https://github.com/ROBOTIS-GIT/turtlebot3_simulations/issues/75
      acc_lim_x: 2.5
      acc_lim_y: 0.0
      acc_lim_theta: 3.2
      decel_lim_x: -2.5
      decel_lim_y: 0.0
      decel_lim_theta: -3.2
      vx_samples: 20
      vy_samples: 5
      vtheta_samples: 20
      sim_time: 1.7
      linear_granularity: 0.05
      angular_granularity: 0.025
      transform_tolerance: 0.2
      trans_stopped_velocity: 0.25
      short_circuit_trajectory_evaluation: True
      stateful: True
      critics: ["RotateToGoal", "Oscillation", "BaseObstacle", "GoalAlign", "PathAlign", "PathDist", "GoalDist"]
      BaseObstacle.scale: 0.02
      PathAlign.scale: 0.0
      GoalAlign.scale: 0.0
      PathDist.scale: 32.0
      GoalDist.scale: 24.0
      RotateToGoal.scale: 32.0
      RotateToGoal.slowing_factor: 5.0
      RotateToGoal.lookahead_time: -1.0

local_costmap:
  local_costmap:
    ros__parameters:
      # Whether to send full costmap every update, rather than updates. (default: false)
      always_send_full_costmap: True

      # Costmap update frequency. (default 5.0)
      update_frequency: 10.0

      # Frequency to publish costmap to topic. (default 1.0)
      publish_frequency: 10.0

      # Reference frame. (default: odom)
      global_frame: map

      # Robot base frame. (default: base_link)
      robot_base_frame: agent_3

      # Whether costmap should roll with robot base frame. (default: False)
      rolling_window: True

      # Height of costmap (m). (default: 5)
      width: 4

      # Width of costmap (m). (default: 5)
      height: 4

      # Resolution of 1 pixel of the costmap, in meters. (default: 0.1)
      resolution: 0.05

      # Robot radius to use, if footprint coordinates not provided. (default: 0.1)
      robot_radius: 0.2

      # List of mapped plugin names for parameter namespaces and names. Note: Each plugin namespace
      # defined in this list needs to have a plugin parameter defining the type of plugin to be
      # loaded in the namespace.
      plugins: ["static_layer", "inflation_layer"]

      inflation_layer:
        plugin: "nav2_costmap_2d::InflationLayer"
        cost_scaling_factor: 2.5
        inflation_radius: 0.3

      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        # https://navigation.ros.org/configuration/packages/costmap-plugins/static.html

        # QoS settings for map topic. (default: True)
        map_subscribe_transient_local: True

map_server:
  ros__parameters:
    use_sim_time: True
    yaml_filename: "turtlebot3_world.yaml"
    save_map_timeout: 5.0